test/suite/Annotations.test.php

```diff
@@ -585,12 +592,72 @@ protected function testGetAnnotationsFromTrait()
             return;
         }
 
-        eval('trait TestTrait { }');
-        $this->setExpectedException(
-            self::ANNOTATION_EXCEPTION,
-            "Reading annotations from interface/trait 'TestTrait' is not supported"
-        );
-        Annotations::ofClass('TestTrait', '@note');
+        $annotations = Annotations::ofClass('SimpleTrait', '@note');
+
+        $this->check(count($annotations) === 1, 'SimpleTrait has one note annotation.');
+    }
+
+    protected function testCanGetMethodAnnotationsIncludedFromTrait()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod(new \ReflectionClass('SimpleTraitTester'), 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class reflection and method name');
+
+        $annotations = Annotations::ofMethod(new \ReflectionMethod('SimpleTraitTester', 'runFromTrait'));
+        $this->check(count($annotations) > 0, 'from method reflection');
+
+        $annotations = Annotations::ofMethod(new SimpleTraitTester(), 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class object and method name');
+
+        $annotations = Annotations::ofMethod('SimpleTraitTester', 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class name and method name');
+    }
+
+    protected function testHandlesMethodAliasingWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('mindplay\test\traits\NamespaceC\TraitTester', 'run');
+        $this->check($annotations[0]->note === 'trait-test-base', 'parent-inherited annotation goes first');
+        $this->check($annotations[1]->note === 'second-test-trait', 'trait-inherited annotation goes second');
+        $this->check($annotations[2]->note === 'trait-tester', 'non-inherited annotation goes third');
+
+        $annotations = Annotations::ofMethod('mindplay\test\traits\NamespaceC\TraitTester', 'conflict');
+        $this->check($annotations[0]->note === 'first-test-trait', 'correct trait is used when insteadof specified');
+
+        $annotations = Annotations::ofMethod('mindplay\test\traits\NamespaceC\TraitTester', 'conflictResolved');
+        $this->check($annotations[0]->note === 'second-test-trait', 'correct trait is used when aliased with as');
+
+        $annotations = Annotations::ofMethod('mindplay\test\traits\NamespaceC\TraitTester', 'runAgain');
+        $this->check($annotations[0]->note === 'trait-test-base', 'aliased methods still inherit from parent');
+        $this->check($annotations[1]->note === 'simple-trait', 'correct trait is used when aliased with as');
+    }
+
+    protected function testCanGetPropertyAnnotationsIncludedFromTrait()
```

Comment:
Property definitions can be identical, but their DocBlocks and therefore annotations can be different. Surely it's not random thing what is used class, parent class or trait property. Most likely properties work the same way as methods (inherited from parent class can be replaced by trait one, but defined right in class where trait is used isn't overwritten). Behavior isn't documented however.

You can create test case and see what happens.